Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/nhin/deferred/request/proxy/NhinPatientDiscoveryDeferredReqProxyWebServiceSecuredImpl.java

```diff
@@ -86,24 +88,18 @@ public MCCIIN000002UV01 respondingGatewayPRPAIN201305UV02(PRPAIN201305UV02 reque
                     response = (MCCIIN000002UV01) client.invokePort(RespondingGatewayDeferredRequestPortType.class,
                             "respondingGatewayDeferredPRPAIN201305UV02", request);
                 } else {
-                    ackMessage = "Failed to call the web service ("
-                            + NhincConstants.PATIENT_DISCOVERY_DEFERRED_REQ_SERVICE_NAME + ").  The URL is null.";
-                    response = HL7AckTransforms.createAckErrorFrom201305(request, ackMessage);
-                    LOG.error(ackMessage);
+                    throw new WebServiceException("Could not determine URL for Patient Discovery Deferred Request endpoint");
                 }
             } else {
-                ackMessage = "Failed to call the web service ("
-                        + NhincConstants.PATIENT_DISCOVERY_DEFERRED_REQ_SERVICE_NAME
-                        + ").  The input parameter is null.";
-                response = HL7AckTransforms.createAckErrorFrom201305(request, ackMessage);
-                LOG.error(ackMessage);
+                throw new IllegalArgumentException("Request Message must be provided");
             }
         } catch (Exception e) {
             ackMessage = "Failed to call the web service ("
                     + NhincConstants.PATIENT_DISCOVERY_DEFERRED_REQ_SERVICE_NAME
                     + ").  An unexpected exception occurred.";
             response = HL7AckTransforms.createAckErrorFrom201305(request, ackMessage);
             LOG.error(ackMessage + "  Exception: " + e.getMessage(), e);
+            throw new ErrorEventException(e, response, "Unable to call Nhin Patient Discovery Deferred Request");
```

Comment:
3-ackMessage get consolidated into one-ackMessage for the response are we ok with this?